### PR TITLE
WRONG_NAME_OF_VARIABLE_INSIDE_ACCESSOR false positive trigger

### LINKED ISSUE #940: WRONG_NAME_OF_VARIABLE_INSIDE_ACCESSOR false positive trigger
## Describe the bug
See e.g. the following code:
```
val String.blaBla: String
    get() = "bla".blaBla("bla")

fun String.blaBla(string: String): String = this + string
```

WRONG_NAME_OF_VARIABLE_INSIDE_ACCESSOR is triggered, saying that `field` keyword should be used instead of property name inside property accessors. But it seems that it mistakes the `blaBla` method name for the property name because they are the same in this case.

## Expected behavior
WRONG_NAME_OF_VARIABLE_INSIDE_ACCESSOR shouldn't be triggered when there's a method with the same name as the property

## Observed behavior
WRONG_NAME_OF_VARIABLE_INSIDE_ACCESSOR is triggered when there's a method with the same name as the property

## Steps to Reproduce
1. Create an extension variable;
2. Call a method, the name of which is the same as the property itself, within the property getter;
3. Run `diktatCheck` to observe the supposed false positive trigger of WRONG_NAME_OF_VARIABLE_INSIDE_ACCESSOR.

## Environment information
* diktat version: 0.6.2
* build tool (maven/gradle): Gradle 6.7
* how is diktat run (CLI, plugin, etc.): gradle plugin
* kotlin version: 1.5.10
* operating system: Windows 10
* link to a project (if your project is public):

### PR DESCRIPTION
**Pull request by Cheshiriks**
### What's done:
* fixed false positive trigger
Closes #940

### CODE REVIEW

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter6/PropertyAccessorFieldsWarnTest.kt [hunk 1]
```diff
@@ -117,4 +117,21 @@ class PropertyAccessorFieldsWarnTest : LintTestBase(::PropertyAccessorFields) {
             LintError(24, 4, ruleId, "${WRONG_NAME_OF_VARIABLE_INSIDE_ACCESSOR.warnText()} set(value) {...")
         )
     }
+
+    @Test
+    @Tag(WarningNames.WRONG_NAME_OF_VARIABLE_INSIDE_ACCESSOR)
+    fun `shouldn't be triggered when there's a method with the same name as the property`() {
```
Comment (kgevorkyan) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter6/PropertyAccessorFieldsWarnTest.kt:123]: Please extend tests, such cases could exist as for function extension
and for class methods
```kotlin
val String.blaBla: String
    get() = "bla".blaBla("bla")


fun String.blaBla(string: String): String = this + string

class A {
    val foo: Int
        get() = foo()

    fun foo(): Int = 15
}

val A.foo: Int
    get() = 1
```

Comment (kgevorkyan) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter6/PropertyAccessorFieldsWarnTest.kt:123]: Also as you check for now all functions from file, make sure, that such case is valid 
```kotlin
class A {

   val String.blaBla: String
  get() = "bla".boo("bla")
 
   fun boo(string: String): String = string + string
}
```

**Review by petertrr — CHANGES_REQUESTED** [review]
Please also add check whether the property is an extension property. `field` keyword is unresolvable in this case and this rule shouldn't be executed

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter6/PropertyAccessorFieldsWarnTest.kt (hunk 1 above)
Comment (Cheshiriks) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter6/PropertyAccessorFieldsWarnTest.kt:123]: fixed

**Review by petertrr — APPROVED** [review]

**Pull request merged**